packages/plugin-core/src/commands/GoToSiblingCommand.ts

```diff
@@ -33,65 +35,222 @@ export class GoToSiblingCommand extends BasicCommand<
   }
 
   async execute(opts: CommandOpts) {
+    // Context for error messages
     const ctx = "GoToSiblingCommand";
-    const maybeTextEditor = VSCodeUtils.getActiveTextEditor();
-    if (!maybeTextEditor) {
+    // Get text editor
+    const textEditor = VSCodeUtils.getActiveTextEditor();
+    if (!textEditor) {
       window.showErrorMessage("You need to be in a note to use this command");
       return {
         msg: "no_editor" as const,
       };
     }
-    let value = "";
-    value = path.basename(maybeTextEditor.document.uri.fsPath, ".md");
-    let respNodes: NoteProps[] = [];
-
-    const { vaults, engine, wsRoot } = ExtensionProvider.getDWorkspace();
-    if (value === "root") {
-      const vault = VaultUtils.getVaultByFilePath({
-        vaults,
-        wsRoot,
-        fsPath: maybeTextEditor.document.uri.fsPath,
-      });
-      const rootNode = NoteUtils.getNoteByFnameFromEngine({
-        fname: value,
-        vault,
-        engine,
-      });
-      if (_.isUndefined(rootNode)) {
-        throw new DendronError({ message: "no root node found" });
+    // Get fname of the active note
+    const fname = path.basename(textEditor.document.uri.fsPath, ".md");
+    // Get extension and workspace
+    const ext = ExtensionProvider.getExtension();
+    const workspace = ext.getDWorkspace();
+    // Get the active note
+    const note = await this.getActiveNote(workspace.engine, fname);
+    if (!note) throw new Error(`${ctx}: ${UNKNOWN_ERROR_MSG}`);
+    // Get the sibling note
+    let siblingNote: NoteProps;
+    // If the active note is a journal note, get the sibling note based on the chronological order
+    if (await this.canBeHandledAsJournalNote(note, workspace.engine)) {
+      const { data, error } = await this.getSiblingForJournalNote(
+        workspace.engine.notes,
+        note,
+        opts.direction
+      );
+      if (error) {
+        VSCodeUtils.showMessage(MessageSeverity.WARN, error.message, {});
+        return { msg: error.name } as CommandOutput;
       }
-      respNodes = rootNode.children
-        .map((ent) => engine.notes[ent])
-        .concat([rootNode]);
+      siblingNote = data.sibling;
```

Comment:
Although I specified `RespV3` as the return type of `getSiblingForJournalNote`, typescript cannot narrow down the type after `if (error)` and thinks `data` might be null.

In the VSCode editor, there is no warning about this type issue (even after I restarted ts server).

any idea why this might be happening?

[build fail message](https://github.com/dendronhq/dendron/runs/7240484471?check_suite_focus=true#step:10:78)

Reply:
typescript isn't very smart about destructuring. for type narrowing to work, you can't destructure the argument
  - bad
  ```ts
  const {error, data} = someFunc
  if (error) {
    throw 
  }
  // COMPILE ERROR
  data.value
  ```
  - good
  ```ts
  const resp = someFunc
  if (resp.error) {
    throw 
  }
  resp.data.value
  ```